sources/models/src/lib.rs

```diff
@@ -121,6 +122,11 @@ struct KubernetesSettings {
     node_labels: HashMap<KubernetesLabelKey, KubernetesLabelValue>,
     node_taints: HashMap<KubernetesLabelKey, KubernetesTaintValue>,
     static_pods: HashMap<Identifier, StaticPod>,
+    bootstrap_token: KubernetesBootstrapToken,
+
+    // Settings with useful default values that can be overridden.
```

Comment:
nit: I think these headers are getting confusing the more settings we add, because we don't "require the user to specify" bootstrap token, it's just for one mode of operation.  Rather than adding another header here and codifying it, I'd vote for getting rid of / changing the first two headers, and keeping / improving the "dynamic settings" one, to make it clear that those are actually the special ones that users probably don't want to set themselves.